src/Storage/StreamWrapper.php

```diff
@@ -0,0 +1,278 @@
+<?php
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Storage;
+
+use Google\Cloud\Exception\GoogleException;
+
+/**
+ * A streamWrapper implementation for handling `gs://bucket/path/to/file.jpg`
+ *
+ * See: http://php.net/manual/en/class.streamwrapper.php
+ */
+class StreamWrapper
+{
+    // Must be public according to the PHP documentation
+    public $context;
+
+    // a GuzzleHttp\Psr7\StreamInterface instance
```

Comment:
This isn't always true, as on line 124 `$this->stream` is set to an instance of `StreamableUploader`